firestore/src/main/firestore_main.cc

```diff
@@ -188,8 +191,17 @@ Settings FirestoreInternal::settings() const {
   const api::Settings& from = firestore_core_->settings();
   result.set_host(from.host());
   result.set_ssl_enabled(from.ssl_enabled());
+
+  // TODO(wuandy): We use the deprecated API for default settings, but mark
+  // `used_legacy_cache_settings_` as false such that new settings API is not
+  // rejected by runtime checks. This should be removed when legacy API is
+  // removed.
+#pragma clang diagnostic push
```

Comment:
What happens with gcc or msvc? Will this fail to build? I think there is a macro in absl to do this in a portable manner. If not, I'm pretty sure there's something in the ios sdk that does this exact thing in a portable manner.

Reply:
Oh yeah, there is one macro from iOS SDK we could use. Thanks.